spec/lib/validation_spec.rb

```diff
@@ -360,14 +360,36 @@
   end
 
   describe "#validate_vcpu_quota" do
+    let(:project) { Project.create(name: "test-quota-project") }
+
+    before do
+      project.add_quota(quota_id: ProjectQuota.default_quotas["VmVCpu"]["id"], value: 10)
+    end
+
     it "sufficient cpu quota" do
-      p = instance_double(Project, current_resource_usage: 5, effective_quota_value: 10, quota_available?: true)
-      expect { described_class.validate_vcpu_quota(p, "VmVCpu", 2) }.not_to raise_error
+      # In total, one VM occupying five plus a request for two more,
+      # seven total, is less than the fixture total of ten.
+      Vm.create_with_id(
+        Vm.generate_uuid,
+        unix_user: "ubi", public_key: "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIGWmPgJE", name: "test-vm", family: "standard", cores: 1,
+        vcpus: 5, cpu_percent_limit: 100, cpu_burst_percent_limit: 100, memory_gib: 4,
+        arch: "x64", boot_image: "ubuntu-jammy", display_state: "running", ip4_enabled: true,
+        created_at: Time.now, location_id: Location::HETZNER_FSN1_ID, project_id: project.id
+      )
+      expect { described_class.validate_vcpu_quota(project, "VmVCpu", 2) }.not_to raise_error
```

Comment:
`expect{ ... }.not_to raise_error` is in general a bad pattern. You should run `...`, and if it raises an error, then the spec fails. Maybe the only reason this is done is due to rubocop/rspec restrictions that a spec must have an expectation, but IMO, we would be better off disabling that, or adding a helper method like `force_expectation!` that did `expect(true).to be true` or something and calling that.

Reply:
See what you think of my fixup. It's a more complex test but also couples the two cases together neatly, I think, whereas the two could have drifted otherwise.